tests/testthat/_snaps/pluralization.md

```diff
@@ -176,3 +176,96 @@
     Output
       9 word
 
+# issue 701
+
+    Code
+      print(pluralize("{NA} file{?s} expected"))
+    Output
+      NA file expected
+    Code
+      print(pluralize("{NA_character_} file{?s} expected"))
+    Output
+      NA file expected
+    Code
+      print(pluralize("{NA_real_} file{?s} expected"))
+    Output
+      NA files expected
+    Code
+      print(pluralize("{NA_integer_} file{?s} expected"))
+    Output
+      NA files expected
+    Code
+      print(pluralize("{NaN} file{?s} expected"))
+    Output
+      NaN files expected
+    Code
+      print(pluralize("{Inf} file{?s} expected"))
+    Output
+      Inf files expected
+    Code
+      print(pluralize("{-Inf} file{?s} expected"))
+    Output
+      -Inf files expected
+
+---
+
+    Code
+      print(pluralize("Found {NA} director{?y/ies}."))
+    Output
+      Found NA directory.
+    Code
+      print(pluralize("Found {NA_character_} director{?y/ies}."))
+    Output
+      Found NA directory.
+    Code
+      print(pluralize("Found {NA_real_} director{?y/ies}."))
+    Output
+      Found NA directories.
+    Code
+      print(pluralize("Found {NA_integer_} director{?y/ies}."))
+    Output
+      Found NA directories.
+    Code
+      print(pluralize("Found {NaN} director{?y/ies}."))
+    Output
+      Found NaN directories.
+    Code
+      print(pluralize("Found {Inf} director{?y/ies}."))
+    Output
+      Found Inf directories.
+    Code
+      print(pluralize("Found {-Inf} director{?y/ies}."))
+    Output
+      Found -Inf directories.
+
+---
+
+    Code
+      print(pluralize("Will remove {?no/the/the} {NA} package{?s}."))
+    Output
+      Will remove the NA package.
+    Code
+      print(pluralize("Will remove {?no/the/the} {NA_character_} package{?s}."))
+    Output
+      Will remove the NA package.
+    Code
+      print(pluralize("Will remove {?no/the/the} {NA_real_} package{?s}."))
+    Output
+      Will remove the NA packages.
```

Comment:
Why is this different from the other two outputs above?

```suggestion
      Will remove the NA package.
```

Reply:
There is earlier code that distinguishes between numeric and non-numeric values. cli assumes that numeric values are a quantity while everything else is a vector where the length is the quantity. Seemed safer to keep that logic at that level even if the behavior is a bit weird.

Reply:
This also is a bit weird based on the behavior noted in #717, because a vector with an NA anywhere gets collapsed to just `NA` by `ansi_collapse()` which makes things even more confusing.